lib/ansible/inventory/manager.py

```diff
@@ -161,13 +165,59 @@ def __init__(self, loader, sources=None, parse=True, cache=True):
         else:
             self._sources = sources
 
+        self._launch_ssh_agent()
+
         # get to work!
         if parse:
             self.parse_sources(cache=cache)
 
         self._cached_dynamic_hosts = []
         self._cached_dynamic_grouping = []
 
+    def _launch_ssh_agent(self):
+        ssh_agent_cfg = C.config.get_config_value('SSH_AGENT')
+        match ssh_agent_cfg:
+            case 'none':
+                return
+            case 'auto':
+                ssh_agent_dir = os.path.join(C.DEFAULT_LOCAL_TMP, 'ssh_agent')
+                os.mkdir(ssh_agent_dir, 0o700)
```

Comment:
Could you move this after the `ssh-agent` lookup? Parts of the path construction are there already. Plus, this would allow skipping the directory creation when lookup failed.

Additionally, if possible, I'd recommend using `pathlib.Path` in place of strings (and only convert to `str` when passing it into external things).